index.html

```diff
@@ -3,47 +3,294 @@
+      var VictimsPercentage = d.Victims / MAX_VICTIMS;
+      bar.setAttribute('style', 'width: ' + (VictimsPercentage * 100) + '%;');
+      // I've broken this operation up into two steps, which is a little different than we did in class.
+      // The end result is the same, though. If 'enrollmentPercentage' equals 0.75, this set the style
+      // attribute of our DIV so that it looks something like <div style="width: 75%;"></div>
+      barWrapper.appendChild(bar);
+      row.appendChild(barWrapper);
+      tbody.appendChild(row);
+    });
+  }
+  // The function declaration above just creates the function, it doesn't yet run it. That's what we
+  // do here, to make sure that the table is populated prior to the user clicking anything.
+ 
+  updateTable();
+  // This is creating an event listener, a function that the browser will call every time the specified
+  // event is "fired" by the selected element (here the button with the id 'sort-asc').
+  document.getElementById('sort-asc').addEventListener('click', function () {
+    // Disable the sort if 'asc' was previously selected, otherwise set the sort to 'asc'
+    if (options.sorted === 'asc') {
+      options.sorted = false;
+    } else {
+      options.sorted = 'asc';
+    }
+    // And now we need to re-run the function that populates the table. Note that the results will be
+    // different because we've change the options object.
+    updateTable();
+  });
+  document.getElementById('sort-desc').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+    if (options.sorted === 'desc') {
+      options.sorted = false;
+    } else {
+      options.sorted = 'desc';
+    }
+    updateTable();
+  });
+  // This is where we'll need to create new event listeners for our filter and next/previous page buttons.
+        document.getElementById('filter-r1').addEventListener('click', function () {
+      if (options.filtered === 'Region 1') {
+        data = conflict.filter(function (d) { return d.Region === "1" })
+    document.getElementById('filter-r2').addEventListener('click', function () {
+    } else if (options.filtered === 'Region 2') {
+      data = conflict.filter(function (d) { 
+        return d.Region === "2"})
+    } else if (options.filtered === 'Region 3') {
+      data = conflict.filter(function (d) { 
+        return d.Region === "3"})
+    } else if (options.filtered === 'Region 4') {
+      data = conflict.filter(function (d) { 
+        return d.Region === "4"})
+    } else if (options.filtered === 'Region 5') {
+      data = conflict.filter(function (d) { 
+        return d.Region === "5"}
+
+  }
+
+
+  // They'll look very similar to the above, just listening to different elements, and perhaps performing
+  // different actions when clicked.
+</script>
+<div class="auxparagraph">
+             <p> Source: Special Unit of Victims - Colombian Presidency </p>
```

Comment:
This works fine, but it's a best practice to have the scripts be the last thing on the page. This is because when the browser will draw the page in order, stopping to process each thing that it encounters. Here, our script is small and inline, so this isn't a big issue, but soon we'll be linking to a script in an external file. The result would be that this <p> would not be shown on the screen until after the script is fully downloaded.

Also note that there's no closing `</div>` tag.